middleware.ts

```diff
@@ -0,0 +1,43 @@
+import { decrypt } from '@/lib/jwt'
+import type { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+
+export async function middleware(request: NextRequest) {
+  // Check if the route is in the dashboard group
+  if (
+    request.nextUrl.pathname.startsWith('/(dashboard)') ||
+    request.nextUrl.pathname.startsWith('/my-account')
+  ) {
+    const session = request.cookies.get('session')?.value
+
+    try {
+      // Verify JWT token
+      const payload = await decrypt(session)
+
+      if (!payload) {
+        // Redirect to home page if no valid session
+        return NextResponse.redirect(new URL('/', request.url))
+      }
+
+      // Valid session, continue to protected route
+      return NextResponse.next()
+    } catch (error) {
+      console.error('Middleware auth error:', error)
+      // Redirect to home page on auth error
+      return NextResponse.redirect(new URL('/', request.url))
+    }
+  }
+
+  // Continue for non-protected routes
+  return NextResponse.next()
```

Comment:
der Pfad ist bereits durch den matcher gesichert

```suggestion
    const session = await getSession()

    try {
      // Verify JWT token
      const payload = await decrypt(session)

      if (!payload) {
        // Redirect to home page if no valid session
        return NextResponse.redirect(new URL('/', request.url))
      }

      // Valid session, continue to protected route
      return NextResponse.next()
    } catch (error) {
      console.error('Middleware auth error:', error)
      // Redirect to home page on auth error
      return NextResponse.redirect(new URL('/', request.url))
    }
```

Reply:
I implemented these suggestions except that we use direct cookie access instead of getSession(). This is because the middleware runs in Edge Runtime, which doesn't support Server Components features (next/headers, MongoDB).